docs/source/en/kv_cache.md

```diff
@@ -343,4 +343,32 @@ print(messages)
 
 ## Re-use Cache to continue generation
 
-Sometimes you would want to fist fill-in cache object with key/values for certain prefix prompt and re-use it several times to generate different sequences from it. We are working hard on adding this feature to 🤗 Transformers and will update this section soon. 
+Sometimes you would want to first fill-in cache object with key/values for certain prefix prompt and re-use it several times to generate different sequences from it. In that case you can construct a `Cache` object that will hold the instruction prompt, and re-use it several times with different text sequences.
+
+```python
+>>> import copy
+>>> import torch
+>>> from transformers import AutoModelForCausalLM, AutoTokenizer, DynamicCache
+
+>>> model_id = "meta-llama/Llama-2-7b-chat-hf"
+>>> model = AutoModelForCausalLM.from_pretrained(model_id, torch_dtype=torch.bfloat16, device_map="cuda")
+>>> tokenizer = AutoTokenizer.from_pretrained(model_id)
+
+>>> INITIAL_PROMPT = "You are a helpful assistant. "
+>>> prompt_cache = DynamicCache()
```

Comment:
I would rather promote the use of Static here but both are fine!

Reply:
Oh, actually I recently found this doesn't work, the `copy.deepcopy` fails even after making it a torch Module. @gante confirmed that it's expected to fail, so I'll remove this section

Reply:
yeah we need to fix the copying issue (and add a test!)

Reply:
Done, the reason was that model forward should be run without grad, otherwise the key/values are non-leaf tensors. Fixed the example and verified it runs

Reply:
Let me know if you have any comments, otherwise will merge

Reply:
On cache reuse (copying a cache object): https://github.com/huggingface/transformers/pull/33297

The problem is a bit more complex that `no_grad` on some cache classes :P

Reply:
Mmm, does that mean the current `deepcopy` doesn't copy all tensors from the list when we use dynamic cache?

Reply:
dynamic cache should be okay with `no_grad` at the moment 🤗 other caches, however, have objects that can't be copied (e.g. in the offloaded caches, the cuda stream can't be copied)

I'm writing a PR that lifts the `no_grad` requirement and handles the other corner cases. And, more importantly, adds tests!